text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,799 @@
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the [rules around merging Resource across schema-url will be dropped](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it. SDKs will no longer fill out SchemaURL on Resource. Additionally, as no (non-service) Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent concerns of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
+  - An OOTB ["Env Variable Entity Detector"](#environment-variable-detector) will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Coordinator would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK.  For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod.   SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
+
+#### Entity Merging and Resource
+
+The most important aspect of this design is how Entities will be merged to construct a Resource.
+
+We provide a simple algorithm for this behavior:
+
+- Construct a set of detected entities, `E`
+  - All entity detectors are sorted by priority (highest first)
+  - For each entity detector `D`, detect entities
+    - For each entity detected, `d'`
+      - If an entity `e'` exists in `E` with same entity type as `d'`, do one of the following:
+        - If the entity identiy and schema_url are the same, merge the descriptive attributes of `d'` into `e'`:
+          - For each descriptive attribute `da'` in `d'`
+            - If `da'.key` does not exist in `e'`, then add `da'` to `ei`
+            - otherwise, ignore.
+        - If the entity identity is the same, but schema_url is different: drop the new entity `d'`
+          *Note: We could offer configuration in this case*
```

Comment:
Do you envision entity schemas with comparable versions? If so, wouldn't we expect the highest version to win?

Reply:
We have not specified comparable versions yet, only equality comparison.

YES I think we should work towards comparable versions, but this is written against the spec we have now. Comparable schema URL would be a different OTEP.

Reply:
In that case, I'd simply add a note, that this is the case and the change in outcome when switching is anticipated and considered acceptable.

Reply:
It's already in the specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/README.md#schema-version-number

When I talk about comparable schema URL - I was referencing *compatibility* (i.e understanding breaking changes).  We can ensure highest wins, however that is not the algorithm I've defined here.

This algorithm is naive and consistent.  Without understanding *compatibility* of version (just ordering), we always prefer the first entity we see in the priority list of entity detectors.  If we see more than one entity we cannot assume the version is important, just the order users defined.   This is the behavior they have today and what we'll preserve.

Reply:
@jsuereth there should be a statement, that, at the moment, it is a goal to provide only solutions that do not require looking into schema_urls. That would be part of a "key assumptions" section close to the beginning of the document. However, I'm not sure if making such a change in the current stage is a good idea as it would result in a restructuring of the entire document.